docs/specification/2025-03-26/basic/authorization.mdx

```diff
@@ -63,119 +66,79 @@ audience. For instance:
 
-- MCP clients _MUST_ follow the OAuth 2.0 Authorization Server Metadata protocol defined
-  in [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414).
-- MCP server _SHOULD_ follow the OAuth 2.0 Authorization Server Metadata protocol.
-- MCP servers that do not support the OAuth 2.0 Authorization Server Metadata protocol,
-  _MUST_ support fallback URLs.
+MCP clients **MUST** follow the OAuth 2.0 Authorization Server Metadata protocol defined
+  in [RFC8414](https://datatracker.ietf.org/doc/html/rfc8414) to obtain the information
+  required to interact with the authorization server.
 
-The discovery flow is illustrated below:
+#### 2.3.3 MCP specific headers for discovery
 
-```mermaid
-sequenceDiagram
-    participant C as Client
-    participant S as Server
-
-    C->>S: GET /.well-known/oauth-authorization-server
-    alt Discovery Success
-        S->>C: 200 OK + Metadata Document
-        Note over C: Use endpoints from metadata
-    else Discovery Failed
-        S->>C: 404 Not Found
-        Note over C: Fall back to default endpoints
-    end
-    Note over C: Continue with authorization flow
-```
-
-#### 2.3.1 Server Metadata Discovery Headers
-
-MCP clients _SHOULD_ include the header `MCP-Protocol-Version: <protocol-version>` during
+MCP clients **SHOULD** include the header `MCP-Protocol-Version: <protocol-version>` during
 Server Metadata Discovery to allow the MCP server to respond based on the MCP protocol
 version.
 
 For example: `MCP-Protocol-Version: 2024-11-05`
 
-#### 2.3.2 Authorization Base URL
-
-The authorization base URL **MUST** be determined from the MCP server URL by discarding
-any existing `path` component. For example:
-
-If the MCP server URL is `https://api.example.com/v1/mcp`, then:
+#### 2.3.4 Sequence Diagram
+The following diagram outlines an example flow:
 
-- The authorization base URL is `https://api.example.com`
-- The metadata endpoint **MUST** be at
-  `https://api.example.com/.well-known/oauth-authorization-server`
-
-This ensures authorization endpoints are consistently located at the root level of the
-domain hosting the MCP server, regardless of any path components in the MCP server URL.
+```mermaid
+sequenceDiagram
+    participant C as Client
+    participant M as MCP Server (Resource Server)
+    participant A as Authorization Server
 
-#### 2.3.3 Fallbacks for Servers without Metadata Discovery
+    C->>M: MCP Request without token
+    M->>C: HTTP 401 Unauthorized with WWW-Authenticate header
+    Note over C: Extract authorization_server from WWW-Authenticate
 
-For servers that do not implement OAuth 2.0 Authorization Server Metadata, clients
-**MUST** use the following default endpoint paths relative to the authorization base URL
-(as defined in [Section 2.3.2](#232-authorization-base-url)):
+    alt WWW-Authenticate header not present
+        C->>M: GET /.well-known/oauth-protected-resource
+        M->>C: Resource metadata with authorization_server URL
```

Comment:
```suggestion
        M->>C: Resource metadata with authorization server URL
```

The property is actually called `authorization_servers` so I took out the underscore so we can talk about just the single authorization server URL that we expect.